src/content_scripts/front.js

```diff
@@ -112,7 +112,10 @@ function createFront(insert, normal, hints, visual, browser) {
     _actions["getSearchSuggestions"] = function (message) {
         var ret = null;
         if (_listSuggestions.hasOwnProperty(message.url)) {
-            ret = _listSuggestions[message.url](message.response);
+            ret = _listSuggestions[message.url](message.response, {
```

Comment:
Here the response for suggestions is ready, why we still need the `url` and `query`?

Reply:
Normally, a user's `addSearchAlias` callback only gets access to the response data. It doesn't have access to the request URL or original query the user typed in. This information is useful in many cases, for example, in my config I am using it to filter results further inside the callback. For example:

```javascript
addSearchAlias('d', 'example', 'https://example.com/?q=', 's', 'https://example.com/api?q=', function(response, request) {
    const data = JSON.parse(response.text);
    return data.filter((item) => item.indexOf(request.query) !== -1)
});
```

I included `url` as well because it may be useful in other cases.